### PR TITLE
Add a new method that allows for creating enrollments with more options.

### PR DESCRIPTION
**Pull request by jcallaway**
Along with an Options object for specifying all of the parameters available to the call, discussed at https://canvas.instructure.com/doc/api/enrollments.html#method.enrollments_api.create

Our use case for this is we want to automatically enroll students in our Canvas courses without the students having to accept an invitation (which is the default).

### CODE REVIEW

**Comment by kawesome** [comment]
Can one of the admins verify this patch?

**Comment by ToeBee** [comment]
I think this change overlaps with some of #54 that just got opened yesterday too. I'll have to sort it out once the merge conflict gets resolved on that one.

**Comment by jcallaway** [comment]
Thanks for taking a look at this.  I looked at #54 and it pretty much does what I need, and better, frankly.  I was unaware of the BaseCanvasModel.toPostMap() method, and passing in an Enrollment object to the method feels cleaner to me than creating an EnrollUserOptions as I do here.

There is one bit I need that's not in #54, however: the courseSectionId property of Enrollment needs to be marked with a @CanvasField annotation, since we will need to specify that.  But I can add that once #54 is merged.

In short, feel free to close this PR and merge in #54 instead.

**Comment by ToeBee** [comment]
Closing this PR since I just merged #54. Will try to get a public release out soon.

**Comment by renaedahiya** [comment]
Hi ToeBee. I have a question about the canvas api you may be able to shed some light on our know someone who can. Is there a good place to put this question? Posting it here for now. Question: When we delete a course section in canvas, the sis id for that section does not go away on the canvas side. The side effect of this is this: when we create a section with that sis id at a later date, it links to the previous section! Not what we want. We want a new section. Any suggestions on how to really delete a section as the solution or is there another approach?
Thank you, Renae

**Comment by ToeBee** [comment]
The [Canvas Developers](https://community.canvaslms.com/groups/canvas-developers) group in the Canvas community would probably be a better place to ask this. There is also the #canvas-lms IRC channel on freenode if you are into IRC. 

I thought I had an answer but I don't think it actually works. I'm guessing you are calling the [section creation API call](https://canvas.instructure.com/doc/api/sections.html#method.sections.create) with the `enable_sis_reactivation` option set to `true`? For me if I don't set it or set it to false, it throws an error saying the SIS ID is already in use even if the section has been "deleted". And the section deletion API call doesn't take any parameters so you can't modify its behavior at all. So yeah, I'm not sure there is a solution to this.

Depending on what you are trying to achieve, you could modify the existing section to change the SIS ID (just add something to the end of it maybe?) and then you could create a new section with the correct SIS ID.
